app/mailers/katello/subscription_mailer.rb

```diff
@@ -10,6 +12,12 @@ def subscription_expiry(options)
         @pools = Katello::Pool.readable.expiring_in_days(days_from_now)
       end
 
+      if @pools.any?
```

Comment:
I was testing this and overall it's great. I noticed one quirk that we can easily fix here:

- the subscriptions report represents the expiring subscriptions in a **host-wise manner**
- in my environment I had subscriptions expiring but **none of them were attached to my hosts** and so the report was empty

Improve this by checking if any hosts are subscribed to these pools:
```
@affected_hosts = Host::Managed.with_pools_expiring_in_days(days_from_now) 
```

The code in this class needs to be updated so that the report url and link are only included in the email if there are some `@affected_hosts`. I now think the report link text should be renamed from 'View full report' to 'View a report of the affected hosts' or to indicate more clearly what the full report contains vs the email